pkg/controller/appdefinition/networkpolicy.go

```diff
@@ -0,0 +1,187 @@
+package appdefinition
+
+import (
+	"fmt"
+	v1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	"github.com/acorn-io/acorn/pkg/config"
+	"github.com/acorn-io/acorn/pkg/labels"
+	"github.com/acorn-io/baaah/pkg/router"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"net"
+	"strconv"
+	"strings"
+)
+
+func NetworkPolicy(req router.Request, resp router.Response) error {
+	cfg, err := config.Get(req.Ctx, req.Client)
+	if err != nil {
+		return err
+	} else if *cfg.DisableNetworkPolicies {
+		return nil
+	}
+
+	app := req.Object.(*v1.AppInstance)
+	appNamespace := app.ObjectMeta.Namespace // this is where the AppInstance lives
+	podNamespace := app.Status.Namespace     // this is where the app is actually running
+
+	// first, create the NetworkPolicy for the whole app
+	resp.Objects(&networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      app.Name,
+			Namespace: podNamespace,
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: labels.Managed(app),
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{{
+				From: []networkingv1.NetworkPolicyPeer{{
+					NamespaceSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							labels.AcornAppNamespace: appNamespace,
+						},
+					}},
+				},
+			}},
+			PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+		},
+	})
+
+	ingressNamespace := *cfg.IngressControllerNamespace
+	podCIDR := *cfg.PodCIDR
+	// make sure the podCIDR is valid
+	if podCIDR != "" {
+		_, _, err = net.ParseCIDR(podCIDR)
+		if err != nil {
+			return err
+		}
+	}
+
+	// next, create NetworkPolicies for each container and job in the app that has a published port
+	for containerName, container := range app.Status.AppSpec.Containers {
+		buildNetPolsForContainer(app.Name, containerName, podNamespace, ingressNamespace, podCIDR, container, resp)
+	}
+	for jobName, job := range app.Status.AppSpec.Jobs {
```

Comment:
good to have this, but I am not sure if job will ever have publish ports as it doesn't make sense

Reply:
I am not sure either but I think it is better to create them for published job ports than no to.